packages/client/src/components/Dashboard/TeamMembers.jsx

```diff
@@ -0,0 +1,204 @@
+    setEditMemberId(id);
+    const memberToEdit = teamMembers.find((member) => member.member_id === id);
+    setEditMemberFirstName(memberToEdit.first_name);
+    setEditMemberLastName(memberToEdit.last_name);
+  };
+
+  const handleDeleteMember = async (id) => {
+    const confirmDelete = window.confirm(
+      'Are you sure you want to delete this team member?',
+    );
+    if (confirmDelete) {
+      try {
+        const response = await fetch(`${apiURL()}/teamMembers/${id}`, {
+          method: 'DELETE',
+        });
+        if (!response.ok) {
+          throw new Error('Failed to delete team member');
+        }
+        setTeamMembers((prevMembers) =>
+          prevMembers.filter((member) => member.member_id !== id),
+        );
+      } catch (error) {
+        console.error(error);
+      }
+    }
+  };
+
+  const handleAddMember = async () => {
+    try {
+      const response = await fetch(`${apiURL()}/teamMembers/addMember`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          first_name: newMemberFirstName,
+          last_name: newMemberLastName,
+        }),
+      });
+      if (!response.ok) {
+        throw new Error('Failed to add team member');
+      }
+      fetchTeamMembers();
+      setShowAddModal(false);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleEditSave = async () => {
+    try {
+      const response = await fetch(`${apiURL()}/teamMembers/${editMemberId}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          first_name: editMemberFirstName,
+          last_name: editMemberLastName,
+        }),
+      });
+      if (!response.ok) {
+        throw new Error('Failed to edit team member');
+      }
+      fetchTeamMembers();
+      setEditMemberId(null);
+      setEditMemberFirstName('');
+      setEditMemberLastName('');
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  return (
+    <div className="team-members-container">
+      <div className="team-members-heading-container">
+        <Typography className="teamMembers-heading" variant="h6">
+          Team Members
+        </Typography>
+        <Button
+          type="button"
+          variant="contained"
+          size="small"
+          onClick={() => setShowAddModal(true)}
+        >
+          Add Team Member
+        </Button>
+      </div>
+      <div className="add-member">
+        <AddTeamMemberModal
+          showAddModal={showAddModal}
+          setShowAddModal={setShowAddModal}
+          newMemberFirstName={newMemberFirstName}
+          setNewMemberFirstName={setNewMemberFirstName}
+          newMemberLastName={newMemberLastName}
+          setNewMemberLastName={setNewMemberLastName}
+          handleAddMember={handleAddMember}
+        />
+      </div>
+
+      {editMemberId && (
```

Comment:
This entire if statement could be a separate React component, so it becomes more readable:

```suggestion
      <EditingMember />
      // ....
      
const EditingMember = () => {
  if (!editMemberId) {
    return null;
  }
  
  return (
    <div className="edit-member-container">
          <Typography variant="body1">Edit Member</Typography>
          <div className="edit-member-details">
          ...
  );
}
```